torch/_decomp/decompositions.py

```diff
@@ -4410,12 +4410,20 @@ def should_fold(tensor1: torch.Tensor, tensor2: torch.Tensor, is_out: bool) -> b
 
     t1_shape = t1.shape
     t1_stride = t1.stride()
+
+    import sympy
+
     # Check the contiguous, we can skip the dim with size of 1
     # as aten: https://github.com/pytorch/pytorch/blob/
     # e201460f8aa1510b4c4686627d57b69756c4b916/aten/src/ATen/TensorGeometry.cpp#L17
+    expected_stride = [sympy.Integer(1)]
+    for size in reversed(t1_shape[1:]):
+        expected_stride.append(size * expected_stride[-1])
     return all(
-        st1 == st2 * s2 if s2 != 1 else True
-        for (st1, st2, s2) in zip(t1_stride[:-2], t1_stride[1:-1], t1_shape[1:-1])
+        (isinstance(size, utils.IntWithoutSymInt) and size == 1) or left == right
```

Comment:
Checking `size == 1` is not enough?

Reply:
Yes, in the PreCI, we met a UT which the size seems to be a unbacked int as symbol of `u0` which failed in `_maybe_evaluate_static`.

Reply:
@ezyang Is this the recommended way for checking `size` is concrete and is `1`?

Reply:
No it's not... usually guard_size_oblivious(size == 1) is what you want here

Reply:
Thanks for the comment and change to use `guard_size_oblivious`.